### PR TITLE
[Bug][Vectorized] Fix BE crash with delete condition and enable_storage_vectorization

### LINKED ISSUE #9509: [Bug] BE crash when enable both enable_vectorized_engine and enable_storage_vectorization
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

dev 1.0.1

### What's Wrong?

The statement will hang for a long time and be will crash when execute "select count(*) from lide_test"; 

be.out:

*** SIGSEGV unkown detail explain (@0x0) received by PID 3595 (TID 0x7f7a697cd700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /data/doris-1.x/be/src/common/signal_handler.h:420
 1# 0x00007F7A83CAC400 in /lib64/libc.so.6
 2# doris::segment_v2::SegmentIterator::_init_current_block(doris::vectorized::Block*, std::vector<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>, std::allocator<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn> > >&) at /data/doris-1.x/be/src/olap/rowset/segment_v2/segment_iterator.cpp:767
 3# doris::segment_v2::SegmentIterator::next_batch(doris::vectorized::Block*) at /data/doris-1.x/be/src/olap/rowset/segment_v2/segment_iterator.cpp:923
 4# doris::BetaRowsetReader::next_block(doris::vectorized::Block*) at /data/doris-1.x/be/src/olap/rowset/beta_rowset_reader.cpp:194
 5# doris::vectorized::VCollectIterator::Level1Iterator::next(doris::vectorized::Block*) at /data/doris-1.x/be/src/vec/olap/vcollect_iterator.cpp:264
 6# doris::vectorized::BlockReader::_direct_next_block(doris::vectorized::Block*, doris::MemPool*, doris::ObjectPool*, bool*) at /data/doris-1.x/be/src/vec/olap/block_reader.cpp:176
 7# doris::vectorized::VOlapScanner::get_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /data/doris-1.x/be/src/vec/exec/volap_scanner.cpp:56
 8# doris::vectorized::VOlapScanNode::scanner_thread(doris::vectorized::VOlapScanner*) at /data/doris-1.x/be/src/vec/exec/volap_scan_node.cpp:210
 9# doris::PriorityWorkStealingThreadPool::work_thread(int) at /data/doris-1.x/be/src/util/priority_work_stealing_thread_pool.hpp:135
10# execute_native_thread_routine in /usr/local/service/doris/lib/be/palo_be
11# start_thread in /lib64/libpthread.so.0
12# __clone in /lib64/libc.so.6

### What You Expected?

MySQL [testdb]> select count(*) from lide_test;
+----------+
| count(*) |
+----------+
|        7 |
+----------+
1 row in set (0.01 sec)

### How to Reproduce?

CREATE TABLE `lide_test` (`c1` varchar(190) NOT NULL COMMENT "",`c2` bigint(20) NOT NULL COMMENT "", `c3` varchar(160) NULL COMMENT "" ) ENGINE=OLAP DUPLICATE KEY(`c1`, `c2`) COMMENT "OLAP" DISTRIBUTED BY HASH(`c3`) BUCKETS 3 ;

INSERT INTO `lide_test` VALUES ('abcdef',1,'fjdsajfldjafljdslajfdl'),('abcdef',2,'fjdsajfldjafljdslajfdl'),('abcdef',4,'fjdsajfldjafljdslajfdl'),('abcdef',5,'fjdsajfldjafljdslajfdl');

delete from lide_test where c1 = 'fjdsajfldjafljdslajfdl';

INSERT INTO `lide_test` VALUES ('abcdef',1,'fjdsajfldjafljdslajfdl'),('abcdef',2,'fjdsajfldjafljdslajfdl'),('abcdef',4,'fjdsajfldjafljdslajfdl'),('abcdef',5,'fjdsajfldjafljdslajfdl');

select count(*) from lide_test;
^CCtrl-C -- query killed. Continuing normally.

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #9509

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -797,7 +797,8 @@ void SegmentIterator::_init_current_block(
         auto cid = _schema.column_id(i);
         auto column_desc = _schema.column(cid);
 
-        if (_is_pred_column[cid]) { //todo(wb) maybe we can release it after output block
+        if (_is_pred_column[cid] ||
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:800]: Please add some comment here, explain why i will GT block->columns()

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -815,8 +816,10 @@ void SegmentIterator::_init_current_block(
 void SegmentIterator::_output_non_pred_columns(vectorized::Block* block) {
     SCOPED_RAW_TIMER(&_opts.stats->output_col_ns);
     for (auto cid : _non_predicate_columns) {
-        block->replace_by_position(_schema_block_id_map[cid],
-                                   std::move(_current_return_columns[cid]));
+        auto loc = _schema_block_id_map[cid];
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:819]: std::vector<int> _schema_block_id_map; It is a map from schema column id to column idx in Block, so all loc should less than block->columns(), why do we need add this check?
If we need add this check, any else clause needed?

Comment (HappenLee) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:820]: have add comment

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (HappenLee) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:801]: have add comment in init `_current_return_columns`

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**